src/ESM.sol

```diff
@@ -66,67 +52,26 @@ contract ESM is DSNote {
         require(z <= x);
     }
 
-    // -- admin --
-    function file(bytes32 job, address obj) external auth note {
-        if (job == "end") end = EndLike(obj);
-        if (job == "sun") sun = obj;
-    }
-
-    function file(bytes32 job, uint256 val) external auth note {
-        if (job == "cap") cap = val;
-    }
-
-    // -- state changes --
     function fire() external note {
-        require(state == START && full(), "esm/not-fireable");
+        require(!fired,  "esm/already-fired");
+        require(full(),  "esm/min-not-reached");
 
         end.cage();
 
-        state = FIRED;
+        fired = true;
     }
 
-    function free() external auth note {
-        require(state == START || state == FIRED, "esm/not-freeable");
-
-        state = FREED;
-    }
-
-    function burn() external auth note {
-        require(state == START || state == FIRED, "esm/not-burnable");
-
-        sum   = 0;
-        state = BURNT;
-
-        bool ok = gem.transfer(address(sun), gem.balanceOf(address(this)));
-
-        require(ok, "esm/failed-transfer");
-    }
-
-    // -- user actions --
     function join(uint256 wad) external note {
-        require(state == START, "esm/not-joinable");
+        require(!fired, "esm/already-fired");
 
         gems[msg.sender] = add(gems[msg.sender], wad);
```

Comment:
Btw do we really need `gems` at all? if it will be done offchain we could still get those addresses which sent funds.

Reply:
Seems like having `gems` would make it easier to audit who pledged funds, without having to run a full node, and therefore to ensure that the token redistribution is fair.

Reply:
Yeah true, I was thinking we could use the events or the different txs to get these values, but yeah, having the gems is probably more practical.